core/src/main/java/com/scalar/db/util/retry/Retry.java

```diff
@@ -0,0 +1,53 @@
+package com.scalar.db.util.retry;
+
+import com.scalar.db.util.ThrowableSupplier;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Retry {
```

Comment:
I think either is fine if it is the only retry logic we use in Scalar DB.
If we are having more retry logic in the near future, using resilient4j-retry might be a little better since generalizing this class ends up creating a similar module to resilient4j-retry.